cluster-autoscaler/debuggingsnapshot/debugging_snapshot.go

```diff
@@ -77,6 +81,21 @@ func (s *DebuggingSnapshotImpl) SetUnscheduledPodsCanBeScheduled(podList []*v1.P
 	}
 }
 
+// SetTemplateNode is the setter for TemplateNode
+func (s *DebuggingSnapshotImpl) SetTemplateNode(templates []*framework.NodeInfo) {
+	if templates == nil {
+		return
+	}
+
+	var TemplateNodeInfo []*framework.NodeInfo
+	for _, template := range templates {
+		t := template.Clone()
+		t.UsedPorts = nil
```

Comment:
Why is this needed? Why do it here, but not in SetNodeGroupInfo?

Reply:
There are fields in that obj which cannot be json marshalled, so they have been removed.
The object being captured in SetNodeGroupInfo is different to the one here. As mentioned above.